### DISCUSSION TITLE
Can submodules work with template repositories?

### DISCUSSION THREAD

**Discussion by CaptainIRS** [community/community#22244] [New to GitHub]
<p>I’ve created a template repository(<a href="https://github.com/CaptainIRS/hacker-plus-template" rel="nofollow noopener">here</a>) which has a git submodule folder in it. But when I try to use the template to create a repository(<a href="https://github.com/c4p74In1r5/test" rel="nofollow noopener">here</a>), even though the <code>.gitmodules</code> file exists as expected, the submodule folder “scripts” from the template repository seemingly vanished somehow, and even if the derived repository is cloned, git commands like <code>git submodule init</code> or <code>git submodule update</code> don’t work.</p>
<p>I’m not really sure about what could be expected in this scenario as I couldn’t find any relevant docs. Ideally, I would expect the link to the submodule repository to be retained in the derived repository and the cloning process to initialize the submodules automatically, as otherwise, just cloning an ordinary repository and using it becomes an easier process than using a template repository.</p>
<p>So, is it even possible to get submodules to work with template repositories? If so, then I’d like to know how to do that. Thanks in advance!</p>

**Comment by yamiacat** [comment]
<p>Hi <a class="mention" href="/captainirs">@CaptainIRS</a>! :wave: Welcome to the Community!</p>
<p>At the moment submodules aren’t working with template repositories, sorry! In the longer term that functionality is intended to be added.</p>
<p>In the shorter term, we’re planning to edit our docs to make it clear that submodules currently aren’t working with templates.</p>
<p>Sorry for any frustration you may have experienced while trying to get them to work!</p>

**Reply by l-monninger** [reply]
Is it in the roadmap to have the submodules behave like templates as well?

**Reply by letmejustputthishere** [reply]
hey, this post was written almost 4 years ago. yet it still seems git submodules don't work with github templates. can this prioritised please 🥹

**Comment by mbruno-xmos** [comment]
<p>Why not spend the time making it work rather than spend time updating the docs saying it doesn’t? Also it’s been over a month and it doesn’t look like the docs have been updated to reflect this. This is all I can find: <a href="https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-template-repository" rel="nofollow noopener">https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-template-repository</a></p>
<p>How hard is it to add a line saying “note: submodules in template repositories do not currently work as expected”</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/captainirs">@CaptainIRS</a> and <a class="mention" href="/mbruno-xmos">@mbruno-xmos</a></p>
<p>Quick update - Submodules should now be included in template repo clones!</p>

**Reply by letmejustputthishere** [reply]
what about code spaces create from a repository template, that still doesn't work, right?

**Comment by MPParsley** [comment]
<p>Is it possible to add a template to an existing repository (cf a remote/fork)?</p>
